homeassistant/components/incharge/config_flow.py

```diff
@@ -0,0 +1,55 @@
+"""Config flow for growatt server integration."""
+from http import HTTPStatus
+from typing import Any
+
+from incharge.api import InCharge
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN, NAME
+
+
+class InChargeConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow class."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialise growatt server flow."""
+        self.data: dict[str, Any] = {}
+        self.api: InCharge | None = None
+
+    @callback
+    def _async_show_user_form(self, errors: dict[str, str] | None = None):
+        """Show the form to the user."""
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Handle the start of the config flow."""
+        if not user_input:
+            return self._async_show_user_form()
```

Comment:
Put `async_show_form` here and reverse the logic so it tests for `user_input` and otherwise shows the form (with or without errors)